### ISSUE TITLE
Error while trying to upload into an empty S3 bucket

### ISSUE THREAD

**Issue by evarga** [lmars/packer-post-processor-vagrant-s3#16]
Starting with an empty S3 bucket the following error is reported (I have tried to upload a Vagrant box for the VirtualBox provider):
```
virtualbox-iso (vagrant-s3): Fetching latest manifest
Build 'virtualbox-iso' errored: 1 error(s) occurred:

* Post-processor failed: The specified key does not exist.
```

**Comment by lmars** [comment]
@evarga that case is supposed to be handled [here](https://github.com/lmars/packer-post-processor-vagrant-s3/blob/master/post-processor.go#L195-L198), but maybe the error message has changed.

I don't have the capacity at the moment to fix this, you should either just create the key first, or try investigating why the error is not being caught.

**Comment by evarga** [comment]
I have found the place where the error is generated, but I am not a Go developer. The problem is that the key can't be created upfront, since the name of the manifest should be dynamic (follow the name of the box). This behavior is even recommended in the documentation.

**Comment by evarga** [comment]
I have managed to solve the bug, albeit never before programmed in Go. This is a proof that it is a superb language! :-)

The error check was wrong. Namely, instead of checking the error message the code should check the AWS error code. When it gets `NoSuchKey` then it is safe to assume that the key is not there.

**Comment by lmars** [comment]
> I have managed to solve the bug, albeit never before programmed in Go. This is a proof that it is a superb language! :-)

Yes Go is quite a simple language, glad you managed to fix this!

Fixed in #17.
